### ISSUE TITLE
Assign a randomly generated value to multiple attributes

### ISSUE THREAD

**Issue by spencer-cheng** [microcks/microcks#621]
Hi, Thanks for the great work for Microcks. I have a user scenario in that I would like to have multiple attributes, say attributeA and attributeB, to be the same randomly generated value. Is it possible to do in Microcks template response? If not, is it possible to add it?

Example response.json
{
  "objA" : {
       "attributeA" : {{randomInt()}}
        ...
  }
  "objB": {
      "attributeB":  **_{{ SAME VALUE as objA.attributeA }}_**
  }
}

Also do we have an expected release date for 1.6?

Thanks & Regards,
Spencer

**Comment by lbroudoux** [comment]
Hi!

Sorry but this is not supported at the moment but it's a great idea!

Maybe we could introduce something like a | notation in order to store the rendered value into a context variable (here `a`):
```json
{
  "objA" : {
    "attributeA" : {{ randomInt() | put(a) }}
  },
  "objB": {
    "attributeB": {{ context.a }}
  }
}
```

What do you think? 

BTW 1.6 release is scheduled for late august. Do not hesitate to star the project if you like it 😉

**Comment by spencer-cheng** [comment]
Thanks for the quick reply. "|  put(NAME)" and "context" sounds great. 
One question from me is the scope of context.  Will there be one context for all services running or it is per file scope if we take AsyncAPI yaml file as an example, or per API scope?  Thanks.

**Comment by lbroudoux** [comment]
I imagine having one context per mock invocation. Context can be used within templates as you suggested or through a SCRIPT dispatcher rule.

**Comment by spencer-cheng** [comment]
Is it possible to support both per mock invocation or under a global context? 
User can choose where to store and read the context variable. I am thinking in some scenarios some mocked APIs may need to return the same value because the APIs may refer to each other in the attribute, for example, ID. But I am not sure about the complexity and whether it violates the definition of random data generation.  Thanks.

**Comment by lbroudoux** [comment]
Hey @spencer-cheng !
I've started implementing some stuff and I'd like to refine things a bit. 
What I propose for now is a context per mock invocation only. Things regarding a mock global dynamic context should still be think about as I imagine different things for simulating statefulness in mocking ...

I've refined the syntax and we should have something like:
```json
{
  "objA" : {
    "attributeA" : {{ randomInt() > put(a) }}
  },
  "objB": {
    "attributeB": {{ a }}
  }
}
```

See the use of `>` as a more standard redirect character like in bash. The access to `a` is also more straightforward without having to identity `context` as we already have a default `context` object attached to the template evaluation. 

What do you think?

**Comment by lbroudoux** [comment]
Now available on branch `1.7.x` and within the `quay.io/microcks/microcks:next` container image 😉
